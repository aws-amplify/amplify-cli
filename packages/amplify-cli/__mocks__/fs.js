const fs = jest.genMockFromModule('fs');

function readdirSync() {
  return ['file1', 'file2'];
}

function unlinkSync() {
  return null;
}

function readFileSync() {
  return {};
}

function statSync() {
  return { isDirectory: () => {} };
}

fs.readdirSync = readdirSync;
fs.readFileSync = readFileSync;
fs.statSync = statSync;
fs.unlinkSync = unlinkSync;

module.exports = fs;
