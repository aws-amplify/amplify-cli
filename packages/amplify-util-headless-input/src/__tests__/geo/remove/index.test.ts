import fs from 'fs-extra';
import path from 'path';
import { validateRemoveGeoRequest } from '../../../index';

const assetRoot = path.resolve(path.join(__dirname, '..', '..', 'assets'));

describe('validate remove geo request', () => {
  it('returns valid payload for map', async () => {
    const rawRequest = fs.readFileSync(path.join(assetRoot, 'geo', 'remove', 'validRequest.map.json'), 'utf8');
    const result = await validateRemoveGeoRequest(rawRequest);
    expect(result).toMatchSnapshot();
  });

  it('rejects promise when invalid payload', async () => {
    const resultPromise = validateRemoveGeoRequest('garbage');
    expect(resultPromise).rejects.toBeTruthy();
  });
});