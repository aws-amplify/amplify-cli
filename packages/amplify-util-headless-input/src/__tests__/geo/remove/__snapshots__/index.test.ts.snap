// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate remove geo request returns valid payload for map 1`] = `
Object {
  "serviceRemoval": Object {
    "name": "mymapname",
    "newDefaultResourceName": "mynewdefaultmap",
    "serviceName": "Map",
  },
  "version": 1,
}
`;
